packages/core-contracts/test/utils/ERC20-EIP2612.test.js

```diff
@@ -0,0 +1,217 @@
+const { ethers } = hre;
+const assert = require('assert');
+const { signERC2612Permit } = require('eth-permit');
+
+const { assertRevert } = require('@synthetixio/core-js/utils/assertions');
+const { findEvent } = require('@synthetixio/core-js/utils/events');
+
+const erc20Test = require('./ERC20.lib');
+
+const MOCK_2612 = 'ERC20_2612Mock';
+
+const NAME = 'Synthetix Network Token';
+const SYMBOL = 'snx';
+const DECIMALS = 18;
+
+// run ERC20 Test Suite
+erc20Test(MOCK_2612);
```

Comment:
Great to have behaviors like this!

Could we tweak the format a bit to follow practices coming from the v2x team?
* Call it behaviors instead of lib, and instead of `erc20Test` call it `itBehavesLikeAnERC20`
* Instead of passing it a string, which represents the name of the contract artifact to test, pass it an instance of a contract (which means that this would need to be moved after the deployment block.